utils/src/main/java/com/cloud/utils/StringUtils.java

```diff
@@ -313,4 +313,42 @@ public static String mapToString(final Map<String, String> map) {
     public static String toCSVList(final List<String> csvList) {
         return join(csvList, ",");
     }
+
+    public static int min(int x, int y, int z)
+    {
+        if (x <= y && x <= z)
+            return x;
+        if (y <= x && y <= z)
+            return y;
+        else
+            return z;
+    }
+
+    /**
+     * Calculates minimum number of edits required to convert from one string to another string.
+     * @param str1
+     *      String that needs editing
+     * @param str2
+     *      Target string that needs after editing
+     * @return minimum number of edits required to convert str1 to str2
+     */
+    public static int minimumEditDistance(String str1, String str2) {
```

Comment:
can we have some unit tests for this, please?
I think it assumes a definition of what an edit is, that we need to make explicit as well.

Reply:
Added unit tests in new commit.